logdevice/lib/ClientImpl.cpp

```diff
@@ -2305,6 +2305,19 @@ ClientImpl::prepareRequest(logid_t logid,
   }
   return req;
 }
+
+// Some compilers (e.g., gcc-7.4.0, the default compiler on Ubuntu 18) may inline calls to prepareRequest<Payload>() in
+// ClientImpl::append() below. If that happens, the compiler will not generate code for prepareRequest<Payload>()
+// instantiation. However, lib/shadow/ShadowClient.cpp requires that instantiation. This leads to link errors when
+// linking examples/ and some tests. Adding an explicit instantiation here to fix.
```

Comment:
@march-github do you mind clang formatting this comment? It's longer that 80 characters.